tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs

```diff
@@ -0,0 +1,54 @@
+use cargo_test_macro::cargo_test;
+use cargo_test_support::{compare::assert_ui, current_dir, file, Project};
+
+use super::{cargo_info, init_registry_without_token};
+
+#[cargo_test]
+fn case() {
+    init_registry_without_token();
+    for ver in ["1.0.0", "2.0.0", "3.0.0"] {
+        cargo_test_support::registry::Package::new("my-package", ver).publish();
+    }
+    // Dep1 depends on 3.0.0, Dep2 depends on 2.0.0, Dep3 depends on 1.0.0
+    cargo_test_support::registry::Package::new("dep1", "1.0.0")
+        .dep("my-package", "3.0.0")
+        .publish();
+    cargo_test_support::registry::Package::new("dep2", "1.0.0")
+        .dep("my-package", "2.0.0")
+        .publish();
+    cargo_test_support::registry::Package::new("dep3", "1.0.0")
+        .dep("my-package", "1.0.0")
+        .publish();
```

Comment:
nit: Might make it easier to read if you align `dep<N>` with `<N>.0.0`

Reply:
Updated.

Reply:
Sorry, I meant with `my-package` version requirement

Reply:
Thank you!